cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -729,30 +730,65 @@ impl ResolutionMessage {
     }
 }
 
-impl From<MessagesItem> for ResolutionMessage {
-    fn from(value: MessagesItem) -> Self {
-        match value {
-            MessagesItem::MessageGeneral(msg) => ResolutionMessage::General(MsgGeneral {
-                level: msg.level,
-                msg: msg.message,
+impl From<ResolutionMessageGeneral> for ResolutionMessage {
+    fn from(r_msg: ResolutionMessageGeneral) -> Self {
+        match r_msg.type_ {
+            MessageType::General => ResolutionMessage::General(MsgGeneral {
+                level: Some(r_msg.level),
+                msg: r_msg.message,
             }),
-            MessagesItem::AttrPathNotFound(msg) => {
+            MessageType::ResolutionTrace => ResolutionMessage::General(MsgGeneral {
+                level: Some(MessageLevel::Trace),
+                msg: r_msg.message,
+            }),
+            MessageType::AttrPathNotFound => {
+                let level = r_msg.level;
+                let msg = r_msg.message;
+                // Should always be present for this type of message, but that's not enforced
+                // by the type system
+                let attr_path = r_msg
+                    .context
+                    .get("attr_path")
+                    .cloned()
+                    .unwrap_or("default_attr_path".into());
```

Comment:
lets create a follow up issue for this

Reply:
Perhaps in the follow up, if there is something unexpected coming back from the service, the default message is shown instead of the client side generated one.  We could expand the message with an annotation that something went wrong, but I'm not sure that's helpful to anyone except us.  Might be something worth sending back via sentry or telemetry though.